### PR TITLE
Change vhost into monopole and firewall into dipole

### PR DESCRIPTION
**Pull request by outscale-nts**

### CODE REVIEW

File: include/packetgraph/firewall.h [hunk 1]
```diff
@@ -39,9 +39,7 @@
  * @param	errp is set in case of an error
  * @return	a pointer to a brick structure on success, NULL on error
  */
-PG_WARN_UNUSED
```
Comment (outscale-jju) [inline include/packetgraph/firewall.h]: Why removing PG_WARN_UNUSED ?

File: include/packetgraph/firewall.h [hunk 2]
```diff
@@ -39,9 +39,7 @@
  * @param	errp is set in case of an error
  * @return	a pointer to a brick structure on success, NULL on error
```
Comment (outscale-jju) [inline include/packetgraph/firewall.h]: Documentation also need update

File: src/firewall.c [hunk 3]
```diff
@@ -205,37 +200,35 @@ static int firewall_burst(struct pg_brick *brick, enum pg_side from,
 
 		pg_low_bit_iterate_full(it_mask, bit, i);
 		tmp = pkts[i];
-
```
Comment (outscale-jju) [inline src/firewall.c]: nope

File: src/firewall.c [hunk 4]
```diff
@@ -205,37 +200,35 @@ static int firewall_burst(struct pg_brick *brick, enum pg_side from,
...
 		/* Firewall only manage IPv4 or IPv6 filtering.
 		 * Let non-ip packets (like ARP) pass.
 		 */
+
```
Comment (outscale-jju) [inline src/firewall.c]: nope

File: src/firewall.c [hunk 5]
```diff
@@ -205,37 +200,35 @@ static int firewall_burst(struct pg_brick *brick, enum pg_side from,
...
 			continue;
 		}
-
 		/* NPF only manage layer 3 so we temporaly cut off layer 2.
 		 * Note that this trick is not thread safe. To do so, we will
 		 * have to clone packets just for filtering and will have to
 		 * restroy cloned packets after handling them in NPF.
 		 */
 		rte_pktmbuf_adj(tmp, sizeof(struct ether_hdr));
-
```
Comment (outscale-jju) [inline src/firewall.c]: nope

File: src/firewall.c [hunk 6]
```diff
@@ -205,37 +200,35 @@ static int firewall_burst(struct pg_brick *brick, enum pg_side from,
...
 		/* filter packet */
 		ret = npf_packet_handler(state->npf,
 					 (struct mbuf **) &tmp,
 					 state->ifp,
 					 pf_side);
 		if (ret)
 			pkts_mask &= ~bit;
-
```
Comment (outscale-jju) [inline src/firewall.c]: nope

File: src/firewall.c [hunk 7]
```diff
@@ -205,37 +200,35 @@ static int firewall_burst(struct pg_brick *brick, enum pg_side from,
...
 		/* Firewall only manage IPv4 or IPv6 filtering.
 		 * Let non-ip packets (like ARP) pass.
 		 */
+
 		eth = rte_pktmbuf_mtod(tmp, struct ether_hdr *);
 		if (eth->ether_type != rte_cpu_to_be_16(ETHER_TYPE_IPv4) &&
 		    eth->ether_type != rte_cpu_to_be_16(ETHER_TYPE_IPv6)) {
 			continue;
 		}
-
```
Comment (outscale-jju) [inline src/firewall.c]: nope

File: tests/firewall/tests.c [hunk 8]
```diff
@@ -657,7 +659,6 @@ static void firewall_noip(enum pg_side dir)
 	pg_brick_destroy(fw);
 	pg_brick_destroy(col);
 }
-
```
Comment (outscale-jju) [inline tests/firewall/tests.c]: nope

File: include/packetgraph/vhost.h [hunk 9]
```diff
@@ -34,9 +34,7 @@ struct pg_brick;
  * @errp:	set in case of an error
  * @return:	a pointer to a brick structure on success, NULL on error
  */
-PG_WARN_UNUSED
```
Comment (outscale-jju) [inline include/packetgraph/vhost.h]: Please, no.

File: src/brick.c [hunk 10]
```diff
@@ -475,7 +475,6 @@ int pg_brick_link(struct pg_brick *west,
 	/* finish the pairing of the edge */
 	pg_brick_get_edge(east, WEST_SIDE, east_index)->pair_index = west_index;
 	pg_brick_get_edge(west, EAST_SIDE, west_index)->pair_index = east_index;
-
```
Comment (outscale-jju) [inline src/brick.c]: nope

File: src/vhost.c [hunk 11]
```diff
@@ -52,6 +52,7 @@ struct pg_vhost_socket {
 
 struct pg_vhost_state {
 	struct pg_brick brick;
+
```
Comment (outscale-jju) [inline src/vhost.c]: nope

File: src/vhost.c [hunk 12]
```diff
@@ -128,6 +126,7 @@ static int vhost_burst(struct pg_brick *brick, enum pg_side from,
 		side->burst_count_cb(side->burst_count_private_data,
 				     bursted_pkts);
 #endif /* #ifdef PG_VHOST_BENCH */
+	rte_atomic32_clear(&state->allow_queuing);
```
Comment (outscale-jju) [inline src/vhost.c]: why ?

File: include/packetgraph/vhost.h [hunk 13]
```diff
@@ -34,9 +34,7 @@ struct pg_brick;
  * @errp:	set in case of an error
  * @return:	a pointer to a brick structure on success, NULL on error
  */
```
Comment (outscale-jju) [inline include/packetgraph/vhost.h]: Documentation not updated

File: src/vhost.c [hunk 14]
```diff
@@ -245,12 +245,18 @@ static int vhost_init(struct pg_brick *brick, struct pg_brick_config *config,
...
-			uint32_t east_max, enum pg_side output,
+static enum pg_side vhost_get_side(struct pg_brick *brick)
+{
+	struct pg_vhost_state *state =
+	pg_brick_get_state(brick, struct pg_vhost_state);
+
+	return pg_flip_side(state->output);
+}
+
+struct pg_brick *pg_vhost_new(const char *name, enum pg_side output,
```
Comment (outscale-jju) [inline src/vhost.c]: We can new remove output as it is defined at link time.

**Comment by outscale-mgo** [comment]
rebase pls

**Pull request closed without merging**